### PR TITLE
fishPlugins.plugin-git: init at 0.1

### PR DESCRIPTION
**Pull request by UnsolvedCypher**
###### Description of changes

This is `plugin-git`: https://github.com/jhillyerd/plugin-git . It is a plugin for the Fish shell with git aliases.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable: (*None applicable*)
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage) (*No dependent packages*)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -15294,6 +15294,11 @@
     github = "unrooted";
     githubId = 30440603;
   };
+  unsolvedcypher = {
```
Comment (cole-h) [inline maintainers/maintainer-list.nix:15297]: Please split this out into its own commit with the message `maintainers: add unsolvedcypher`

Comment (UnsolvedCypher) [reply maintainers/maintainer-list.nix:15297]: Done!

File: pkgs/shells/fish/plugins/default.nix [hunk 2]
```diff
@@ -32,6 +32,8 @@ lib.makeScope newScope (self: with self; {
 
   pisces = callPackage ./pisces.nix { };
 
+  plugin-git = callPackage ./plugin-git.nix { };
```
[the remark this answers is not in the document]
Comment (UnsolvedCypher) [reply pkgs/shells/fish/plugins/default.nix:35]: Good catch, sorry about that. I've added it in

**Review by cole-h — APPROVED** [review]
Thank you for your contribution! Welcome to Nixpkgs!

---

Result of `nixpkgs-review pr 217425` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fishPlugins.plugin-git</li>
  </ul>
</details>
